addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -57,7 +57,17 @@ options.registry.TableOfContent = options.Class.extend({
             $el.attr('id', id);
             $el[0].dataset.anchor = 'true';
         });
-        $nav.find('a:first').addClass('active');
+        const tocAnchorEl = this.$target[0].querySelector('a.table_of_content_link');
+        if (!tocAnchorEl) {
+            this.trigger_up('go_to_parent', {$snippet: this.$target});
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('will_remove_snippet', {$target: this.$target});
+            this.$target[0].remove();
+            document.getElementById('wrap').setAttribute('contenteditable', 'false');
```

Comment:
I don't understand why you don't simply trigger_up remove_snippet instead of doing all that..? Is there a reason ?
It will take care of everything.  The call of `will_remove_snippet`, the `go_to_parent`, the `snippet_removed` which set contenteditable etc...